src/main/java/org/opensearch/ad/ml/EntityColdStarter.java

```diff
@@ -590,18 +597,24 @@ private int calculateColdStartDataSize(List<double[][]> coldStartData) {
      * @return the chosen strideLength and numberOfSamples
      */
     private Pair<Integer, Integer> selectRangeParam(AnomalyDetector detector) {
-        long delta = detector.getDetectorIntervalInMinutes();
         int shingleSize = detector.getShingleSize();
-        int strideLength = defaulStrideLength;
-        int numberOfSamples = defaultNumberOfSamples;
-        if (delta <= 30 && 60 % delta == 0) {
-            strideLength = (int) (60 / delta);
-            numberOfSamples = (int) Math.ceil((shingleSize + numMinSamples) / 24.0d) * 24;
+        if (allowInterpolation) {
+            long delta = detector.getDetectorIntervalInMinutes();
+
+            int strideLength = defaulStrideLength;
+            int numberOfSamples = defaultNumberOfSamples;
+            if (delta <= 30 && 60 % delta == 0) {
+                strideLength = (int) (60 / delta);
+                numberOfSamples = (int) Math.ceil((shingleSize + numMinSamples) / 24.0d) * 24;
+            } else {
+                strideLength = 1;
```

Comment:
nit:
Case number 2 in the algorithm where the interval is above 30 and not divisable by 60 is the same as thew new case with no interpolation. We can perhaps factor this out? I also understand why to keep it this way because they might have the same output but for different reasons. It is just a little confusing as I read the entire comment as whole, and now the "uncommon case" is the default one pretty much.

Reply:
I thought about combining conditions of the uncommon code. It would be

if (!allowInterpolation || (delta > 20 || 60%delta != 0))) {
  // uncommon case
} else {
  ...
}

This looks less readable than current implementation. What do you think?

Reply:
Don't have a strong preference, if you considered both and decided current as more readable then we can keep as is. Since we are now using no interpolation, should we add more to the comment or change anything since it seems like we are taking what was "uncommon" in the past as now to be the default.

Reply:
I added comment "If we disable interpolation, strideLength is 1 and numberOfSamples is shingleSize + numMinSamples;". Anything else you want to see?

Reply:
Just saw the new comments, thanks for adding it

Reply:
Thanks for your suggestion